### PR TITLE
Add support for `__round__` in `Series` and `DataFrame`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #14083 

This PR fixes builtin function `round` call on `DataFrame` & `Series` to work by implementing `__round__` method.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by bdice** [comment]
@galipremsagar Might double check against the other operators listed in https://github.com/rapidsai/cudf/issues/10177, too.

**Comment by galipremsagar** [comment]
> @galipremsagar Might double check against the other operators listed in #10177, too.

Nice ! Is it okay if I add those in a follow-up PR?

File: python/cudf/cudf/core/indexed_frame.py [hunk 1]
```diff
@@ -358,6 +358,9 @@ def _from_columns_like_self(
             override_dtypes=override_dtypes,
         )
 
+    def __round__(self, digits=0):
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cudf/cudf/core/indexed_frame.py:361]: I thought so too, `BinaryOperand` is where you are talking about? One reason I placed this in `IndexedFrame` is beacuse `Index` doesn't have `__round__` implemented in pandas.

[the remark this answers is not in the document]
Comment (bdice) [reply python/cudf/cudf/core/indexed_frame.py:361]: Maybe we can add a comment to indicate that. It’s not immediately obvious why this is the right place to put this method.

**Review by bdice — APPROVED** [review]

**Pull request merged**